### PR TITLE
feat(a11y): add aria invalid/describedby attrs to json schema form

### PR DESCRIPTION
**Pull request by nekolab**
A new prop `errorsDescriedById` was added into `<JsonSchemaForm />` to support rendering `aria-invalid` and `aria-describedby` on form controls.

this fix FTI-5660

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [ ] are not breaking changes.

### Documentation
- [ ] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/core/components/layout-utils.jsx [hunk 1]
```diff
@@ -135,6 +135,9 @@ export class Select extends React.Component {
     className: PropTypes.string,
     disabled: PropTypes.bool,
     description: PropTypes.string,
+
```
Comment (davidma415) [inline src/core/components/layout-utils.jsx:138]:
```suggestion
```
An empty line here I think. Also, can we camelcase these prop names?

File: src/core/components/layout-utils.jsx [hunk 2]
```diff
@@ -190,7 +193,16 @@ export class Select extends React.Component {
     let value = this.state.value?.toJS?.() || this.state.value
 
     return (
-      <select className={this.props.className} multiple={ multiple } value={value} onChange={ this.onChange } disabled={disabled} aria-label={description}>
+      <select
+        className={this.props.className}
+        multiple={ multiple }
+        value={value}
+        onChange={ this.onChange }
```
Comment (davidma415) [inline src/core/components/layout-utils.jsx:200]: small nit: don't think we need the spaces for before / after the prop e.g. they should be like how it's passed in `value`

File: src/core/json-schema-components.jsx [hunk 3]
```diff
@@ -17,6 +17,7 @@ const JsonSchemaPropShape = {
   fn: PropTypes.object.isRequired,
   schema: PropTypes.object,
   errors: ImPropTypes.list,
+  errorsDescriedById: PropTypes.string,
```
Comment (davidma415) [inline src/core/json-schema-components.jsx:20]: nit: typo
```suggestion
  errorsDescribedById: PropTypes.string,
```

File: src/core/json-schema-components.jsx [hunk 4]
```diff
@@ -46,7 +47,7 @@ export class JsonSchemaForm extends Component {
   }
 
   render() {
-    let { schema, errors, value, onChange, getComponent, fn, disabled } = this.props
+    let { schema, errors, errorsDescriedById, value, onChange, getComponent, fn, disabled } = this.props
```
Comment (davidma415) [inline src/core/json-schema-components.jsx:50]:
```suggestion
    let { schema, errors, errorsDescribedById, value, onChange, getComponent, fn, disabled } = this.props
```

File: src/core/json-schema-components.jsx [hunk 5]
```diff
@@ -58,7 +59,7 @@ export class JsonSchemaForm extends Component {
     if (!Comp) {
       Comp = getComponent("JsonSchema_string")
     }
-    return <Comp { ...this.props } errors={errors} fn={fn} getComponent={getComponent} value={value} onChange={onChange} schema={schema} disabled={disabled}/>
+    return <Comp { ...this.props } errors={errors} errorsDescriedById={errorsDescriedById} fn={fn} getComponent={getComponent} value={value} onChange={onChange} schema={schema} disabled={disabled}/>
```
Comment (davidma415) [inline src/core/json-schema-components.jsx:62]:
```suggestion
    return <Comp { ...this.props } errors={errors} errorsDescribedById={errorsDescribedById} fn={fn} getComponent={getComponent} value={value} onChange={onChange} schema={schema} disabled={disabled}/>
```

File: src/core/json-schema-components.jsx [hunk 6]
```diff
@@ -71,7 +72,7 @@ export class JsonSchema_string extends Component {
   }
   onEnumChange = (val) => this.props.onChange(val)
   render() {
-    let { getComponent, value, schema, errors, required, description, disabled } = this.props
+    let { getComponent, value, schema, errors, errorsDescriedById, required, description, disabled } = this.props
```
Comment (davidma415) [inline src/core/json-schema-components.jsx:75]:
```suggestion
    let { getComponent, value, schema, errors, errorsDescribedById, required, description, disabled } = this.props
```

File: src/core/json-schema-components.jsx [hunk 7]
```diff
@@ -81,6 +82,10 @@ export class JsonSchema_string extends Component {
     }
     errors = errors.toJS ? errors.toJS() : []
 
+    const ariaProps = errors.length
+      ? { "aria-invalid": "true", "aria-describedby": errorsDescriedById }
```
Comment (davidma415) [inline src/core/json-schema-components.jsx:86]:
```suggestion
      ? { "aria-invalid": "true", "aria-describedby": errorsDescribedById }
```

File: src/core/json-schema-components.jsx [hunk 8]
```diff
@@ -345,7 +373,8 @@ export class JsonSchema_boolean extends Component {
                     disabled={ disabled }
                     allowedValues={ enumValue }
                     allowEmptyValue={ !required }
-                    onChange={ this.onEnumChange }/>)
+                    onChange={ this.onEnumChange }
+                    { ...ariaProps } />)
```
Comment (davidma415) [inline src/core/json-schema-components.jsx:377]: Can we fix the formatting on this `Select` block? e.g. `/>` and `)` seem to be incorrectly indented

**Review by davidma415 — CHANGES_REQUESTED** [review]
Some nits, thanks for this work!

File: src/core/components/layout-utils.jsx (hunk 1 above)
Comment (nekolab) [reply src/core/components/layout-utils.jsx:138]: This empty line is used to separate the "component props" and "DOM native aria props". Since the `aria-*` series are well-known DOM attributes, would be mapped to the native `<select />` element directly and the props in `<select />` are even kebab-case [at React native level](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d304ee67cd3b0abbf52a032f14e885949466b100/types/react/index.d.ts#L1665), I'd like to keep them kebab-case.

File: src/core/json-schema-components.jsx (hunk 8 above)
Comment (nekolab) [reply src/core/json-schema-components.jsx:377]: I believe that's the code style used by `swagger-ui`, they are still using this style in the upstream code

File: src/core/components/layout-utils.jsx (hunk 2 above)
Comment (nekolab) [reply src/core/components/layout-utils.jsx:200]: It seems we (and the upstream project!) are mixing the usage of these two styles everywhere, Iet me change it here and leave the rest as is 😮‍💨

**Comment by nekolab** [comment]
@davidma415  Need a re-review for [this PR](https://github.com/Kong/swagger-ui-kong-theme/pull/181) too due to the correction of `errorsDescribedById` ❤️

**Review by davidma415 — APPROVED** [review]
lgtm!
